packages/expo-router/_entry.tsx

```diff
@@ -9,7 +9,13 @@ import { getNavigationConfig } from "./src/getLinkingConfig";
 import { getRoutes } from "./src/getRoutes";
 import { loadStaticParamsAsync } from "./src/loadStaticParamsAsync";
 
-export const ctx = require.context(process.env.EXPO_ROUTER_APP_ROOT!);
+export const ctx = require.context(
+  process.env.EXPO_ROUTER_APP_ROOT!,
+  true,
+  /.*/,
+  // @ts-expect-error
+  process.env.EXPO_ROUTER_IMPORT_MODE!
```

Comment:
Ideally, this wouldn't need to change at all between build modes. When splitting (lazy bundling) isn't enabled, `import()`  is effectively just a Promise wrapper around `require`. We'd need to make sure that the `import()` promise tick doesn't cause us to briefly show the loading state (which I *think* Suspense should handle...?)

```suggestion
  'lazy',
```